### ISSUE TITLE
Question about Accessing the Predictions of Each Additive Function

### ISSUE THREAD

**Issue by brian-liu-24** [interpretml/interpret#160]
Hi,
Thanks for putting together such an interesting model! We are currently trying to access the prediction scores of each additive function by using the local feature explanations. However, we found that the sum of these scores do not exactly equal the final prediction, although they are very close. Is there an easy way to access the prediction scores of each additive function in the model directly?
Thank you!

**Comment by interpret-ml** [comment]
Hi @brian-liu-24, 

The local feature explanations are the per-additive-function prediction, and should sum up to the final prediction of the model in the regression setting. In classification, the local explanations are expressed as logits (https://en.wikipedia.org/wiki/Logit). To get a probability in the classification setting, the logits per feature are summed up and passed to the logistic link function (https://en.wikipedia.org/wiki/Logistic_function), exactly like in logistic regression. We have a more comprehensive explanation in this issue: https://github.com/interpretml/interpret/issues/157 .

Is there any chance you omitting the intercept term in your summation? This might explain the small discrepancy you are seeing. You can access the intercept through the ```intercept_``` attribute on the model. Alternatively, you can use the ```.data()``` function of the local explanation objects, which gives you access to the full prediction breakdowns per instance, and also includes the intercept:

```python3
ebm_local = ebm.explain_local(X)
data_dict = ebm_local.data(0)  # Breakdown of prediction for instance 0

# Access breakdowns per feature
for feature_name, contribution, original_value in zip(data_dict[‘names’], data_dict[‘scores’], data_dict[‘values’]):
    print(feature_name, contribution, original_value)

# Access intercept
intercept = data_dict['extra']['scores']
```

The programmatic access through the ```.data()``` function also includes the scores for all features, and not just the top 15 which we visualize inside Jupyter notebooks, which might also explain the small discrepancy you're seeing. Hope this helps, and please feel free to ask more questions!

-InterpretML Team
